static/app/views/performance/browser/resources/resourceSummaryPage/sampleImages.tsx

```diff
@@ -33,51 +44,90 @@ function SampleImages({groupId}: Props) {
-      <ImageWrapper>
-        {filteredResources.map(resource => {
-          return (
-            <ImageContainer
-              src={resource[SPAN_DESCRIPTION]}
-              fileName={getFileNameFromDescription(resource[SPAN_DESCRIPTION])}
-              size={resource[`measurements.${HTTP_RESPONSE_CONTENT_LENGTH}`]}
-              key={resource[SPAN_DESCRIPTION]}
-            />
-          );
-        })}
-      </ImageWrapper>
+    <ChartPanel title={showImages ? t('Largest Images') : undefined}>
+      {showImages ? (
+        <ImageWrapper>
+          {filteredResources.map(resource => {
+            return (
+              <ImageContainer
+                src={resource[SPAN_DESCRIPTION]}
+                showImage={isImagesEnabled}
+                fileName={getFileNameFromDescription(resource[SPAN_DESCRIPTION])}
+                size={resource[`measurements.${HTTP_RESPONSE_CONTENT_LENGTH}`]}
+                key={resource[SPAN_DESCRIPTION]}
+              />
+            );
+          })}
+        </ImageWrapper>
+      ) : (
+        // TODO - the selection to show only links should be persisted for awhile
+        <DisabledImages onClickShowLinks={() => setShowImages(true)} />
+      )}
     </ChartPanel>
   );
 }
 
-function ImageContainer({
-  src,
-  fileName,
-  size,
-}: {
+function DisabledImages(props: {onClickShowLinks?: () => void}) {
+  const {onClickShowLinks} = props;
+  const {
+    selection: {projects: selectedProjects},
+  } = usePageFilters();
+  const {projects} = useProjects();
+  const firstProjectSelected = projects.find(
+    project => project.id === selectedProjects[0].toString()
+  );
+
+  return (
+    <div>
+      <DisabledImageTextContainer>
+        <h6>{t('Images not shown')}</h6>
+        {t(
+          'You know, you can see the actual images that are on your site if you opt into this feature.'
+        )}
+      </DisabledImageTextContainer>
+      <ButtonContainer>
+        <Button onClick={onClickShowLinks}>Only show links</Button>
+        <Link
+          to={normalizeUrl(
+            `/settings/projects/${firstProjectSelected?.slug}/performance/`
+          )}
+        >
+          <Button priority="primary">Enable in Settings</Button>
+        </Link>
+      </ButtonContainer>
+    </div>
+  );
+}
+
+function ImageContainer(props: {
   fileName: string;
+  showImage: boolean;
   size: number;
   src: string;
 }) {
+  const theme = useTheme();
+  const [hasError, setHasError] = useState(false);
+
+  const {fileName, size, src, showImage = true} = props;
   const fileSize = getDynamicText({
     value: formatBytesBase2(size),
     fixed: 'xx KB',
   });
 
+  const commonStyles: CSSProperties = {minWidth: imageWidth, height: '200px'};
+
   return (
     <div style={{width: '100%', wordWrap: 'break-word'}}>
-      <img src={src} style={{minWidth: imageWidth, height: '200px'}} />
+      {showImage && !hasError ? (
+        <img onError={() => setHasError(true)} src={src} style={commonStyles} />
```

Comment:
@AbhiPrasad Do you know if errors from images that fail to load will become issues in our sentry project? As we will expect errors here, we wouldn't want them to report them to sentry. Although something interesting here would be to collect metrics on how often images aren't available or are available, maybe would Javascript DDM??

Reply:
By default we don't collect errors from resource 404s. See: https://blog.sentry.io/fallbacks-for-http-404-images-in-html-and-javascript/ for more info.

> Although something interesting here would be to collect metrics on how often images aren't available or are available, maybe with Javascript DDM

Yes we can totally do that. Every time an image loads we can increment a metric.

```ts
// status is either `ok` or `error` based on if the image was available or not
Sentry.metrics.increment('performance.resource.image_load', 1, { tags: { status } });
```

We need to put the `Sentry.metrics.increment` in a `useEffect` call because it's a side effect.

Reply:
👍 I'll add this in another PR, as it won't be useful until we release.